src/Microsoft.AspNet.Mvc.Core/DependencyInjection/MvcCoreMvcCoreBuilderExtensions.cs

```diff
@@ -62,7 +62,7 @@ public static IMvcCoreBuilder AddFormatterMappings(this IMvcCoreBuilder builder)
         // Internal for testing.
         internal static void AddFormatterMappingsServices(IServiceCollection services)
         {
-            services.TryAddTransient<FormatFilter, FormatFilter>();
+            services.TryAddSingleton<FormatFilter, FormatFilter>();
```

Comment:
`FormatFilter` is a service that is an implementation detail of `FormatFilterAttribute`. `IFormatFilter` is for items that are contributors to the 'format filter' protocol.